app/activity/components/activity-form.tsx

```diff
@@ -0,0 +1,286 @@
+import {
+  Box,
+  FormControl,
+  FormLabel,
+  FormErrorMessage,
+  Button,
+  Image,
+  SimpleGrid,
+  Text,
+  Flex,
+  VStack,
+  Input
+} from '@chakra-ui/react'
+import { PrimaryButton } from '@/components/button'
+import { CustomDateTimePicker } from '@/components/customDateTimePicker'
+import { InputForm, TextareaForm } from '@/components/input'
+import { getDateObj } from '@/libs/utils'
+import { useActivityCreate, useActivityUpdate } from '../hooks'
+import { ActivitySchema, activityResolver } from '../schema'
+
+export type ActivityDetails = {
+  id: string
+  title: string
+  timeFrom?: string | null
+  timeTo?: string | null
+  address?: string | null
+  url?: string | null
+  memo?: string | null
+  cost?: string | null
+  costUnit?: string | null
+  uploadedFileUrls?: string[] | null
+  refetch: () => void
+  refetchLoading: boolean
+}
+
+type ActivityFormProps = {
+  tripId: string
+  activityDetails?: ActivityDetails
+}
+
+export const ActivityForm = ({
+  tripId,
+  activityDetails
+}: ActivityFormProps) => {
+  const [selectedImages, setSelectedImages] = useState<File[]>([])
+  const removeImage = (index: number) => {
+    setSelectedImages((prevImages) => {
+      const newImages = [...prevImages]
+      newImages.splice(index, 1)
+      return newImages
+    })
+  }
+
+  const {
+    register,
+    control,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<ActivitySchema>({
+    defaultValues: {
+      title: activityDetails?.title,
+      timeFrom: activityDetails?.timeFrom
+        ? getDateObj(activityDetails?.timeFrom)
+        : undefined,
+      timeTo: activityDetails?.timeTo
+        ? getDateObj(activityDetails?.timeTo)
+        : undefined,
+      address: activityDetails?.address || undefined,
+      url: activityDetails?.url || undefined,
+      memo: activityDetails?.memo || undefined,
+      cost: activityDetails?.cost || undefined,
+      costUnit: activityDetails?.costUnit || undefined,
+      uploadedFileUrls: activityDetails?.uploadedFileUrls || undefined,
+      newFiles: []
+    },
+    resolver: activityResolver
+  })
+  const newFiles = useWatch({ control, name: 'newFiles' })
+
+  const { createActivity, isActivityCreating } = useActivityCreate(tripId)
+  const { updateActivity, isActivityUpdating } = useActivityUpdate(tripId)
+  const mutateFun = activityDetails
+    ? (data: ActivitySchema) => updateActivity(activityDetails.id, data)
+    : createActivity
+  const isMutating = isActivityCreating || isActivityUpdating
+
+  return (
+    <VStack
+      as="form"
+      onSubmit={handleSubmit(mutateFun)}
+      pt={{ base: '40px', md: '40px' }}
+    >
+      <FormControl isRequired isInvalid={!!errors.title}>
+        <FormLabel>Title</FormLabel>
+        <InputForm
+          type="text"
+          placeholder="Asakusa Temple"
+          {...register('title')}
+        />
+        {errors.title && (
+          <FormErrorMessage>{errors.title.message}</FormErrorMessage>
+        )}
```

Comment:
This does not show if you put empty on this input 👀   
Instead of Chakra Message, the HTML required message showed, could you check it? It is better to show ChakraUI message for consistency.

Reply:
> Instead of Chakra Message, the HTML required message showed, could you check it? It is better to show ChakraUI message for consistency.

Do you mean that a toast message is better than an HTML validation message?
I use FormErrorMessage component of ChakraUI 🤔

Reply:
I mean, this field validation works HTML default original message before showing the ChakraUI message because of `isRequired` props. For this reason, I don't normally use "`sRequired` props, but instead, I customize the `FormLabel` component with `isRequired : boolean` props. 

HTML default required message
<img width="586" alt="Screenshot 2024-02-26 at 12 30 26 PM" src="https://github.com/tabi-memo/frontend/assets/66394413/52415e04-708e-45f1-b475-b0c99dcd0986">

Chakra validation error message (should be this)
<img width="633" alt="Screenshot 2024-02-26 at 12 29 55 PM" src="https://github.com/tabi-memo/frontend/assets/66394413/c97e8830-d166-4977-b557-de26c13fce3c">

Reply:
Fixed by https://github.com/tabi-memo/frontend/pull/34/commits/686850ea26956625e25c0b9ddf1169a40d16592d